addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -183,11 +186,31 @@ options.registry.GalleryLayout = options.registry.CarouselHandler.extend({
             // This is not needed anymore as the images of the rendered
             // `website.gallery.slideshow` are replaced by the elements of
             // `imgHolderEls`.
-            attrClass: imageEls.length > 0 ? imageEls[0].className : '',
             attrStyle: imageEls.length > 0 ? imageEls[0].style.cssText : '',
         },
         $slideshow = $(renderToElement('website.gallery.slideshow', params));
-        const imgSlideshowEls = $slideshow[0].querySelectorAll("img[data-o-main-image]");
+        const videoEls = elements.filter((el) => el.isVideo);
+        for (const element of videoEls) {
+            const videoEl = $slideshow[0].querySelector(`img[src='${element.src}']`);
+            // Todo: To remove this code and adapt in xml code in master
+            // Template: Replace image tag with media_iframe_video
+            const videoContainerHTML = `
+                <div class="d-block media_iframe_video" data-o-main-image="true" data-oe-expression="${element.src}">
+                    <div class="css_editable_mode_display"></div>
+                    <div class="media_iframe_video_size"></div>
+                    <iframe src="${element.src}" frameborder="0" allowfullscreen="allowfullscreen"></iframe>
+                </div>`;
+            videoEl.insertAdjacentHTML("afterend", videoContainerHTML);
```

Comment:
What about putting this inside a static template and using `renderToElement` like above ? (avoiding the security warning)